WarehouseManagement.Core/Services/MarkerService.cs

```diff
@@ -18,29 +21,43 @@ public MarkerService(IRepository repository)
         this.repository = repository;
     }
 
-    public async Task AddAsync(MarkerFormDto model, string userId)
+    public async Task<int> AddAsync(MarkerFormDto model, string userId)
     {
+        if (await ExistByNameAsync(model.Name))
+        {
+            throw new ArgumentException($"{MarkerWithNameExist} {model.Name}");
+        }
+
         var marker = new Marker
         {
             Name = model.Name,
             CreatedAt = DateTime.UtcNow,
             CreatedByUserId = userId,
         };
+
         await repository.AddAsync(marker);
         await repository.SaveChangesAsync();
+
+        return marker.Id;
     }
 
     public async Task DeleteAsync(int id, string userId)
     {
         var marker = await repository.GetByIdAsync<Marker>(id);
+
         if (marker == null)
         {
-            throw new KeyNotFoundException($"Marker with ID {id} not found.");
+            throw new KeyNotFoundException($"{MarkerWithIdNotFound} {id}");
         }
 
-        marker.IsDeleted = true;
-        marker.DeletedAt = DateTime.UtcNow;
-        marker.DeletedByUserId = userId;
+        var usages = await GetMarkerUsagesAsync(id);
+        if (usages.Any())
+        {
+            var usageMessage =
+                $"{MarkerHasUsages} "
+                + string.Join(" ", usages.Select(u => $"{u.Key}: {string.Join(",", u.Value)}"));
+            throw new InvalidOperationException(usageMessage);
+        }
```

Comment:
![image](https://github.com/InterCarsBulgaria-TechLeadAcademy/warehouse.management.be/assets/70769278/5af770c5-feeb-4f81-85c5-e9104ba8db1d)
Group the usages by Type(Zones,Vendors,Deliveries)